gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/jupiter/handlers/api/v4/processor/ApiMessageProcessorChainFactory.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.handlers.api.v4.processor;
+
+import io.gravitee.gateway.jupiter.api.hook.ProcessorHook;
+import io.gravitee.gateway.jupiter.core.processor.Processor;
+import io.gravitee.gateway.jupiter.core.processor.ProcessorChain;
+import io.gravitee.gateway.jupiter.handlers.api.v4.Api;
+import io.gravitee.gateway.jupiter.handlers.api.v4.processor.message.error.SimpleFailureMessageProcessor;
+import io.gravitee.gateway.jupiter.handlers.api.v4.processor.message.error.template.ResponseTemplateBasedFailureMessageProcessor;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ApiMessageProcessorChainFactory {
+
+    private final List<ProcessorHook> processorHooks = new ArrayList<>();
+
+    public ProcessorChain messageProcessorChain(final Api api) {
```

Comment:
pre/post was meant for request/response; in this case, the processor can act on both request message and response message. so in this context, I don't know if we should call it post, because there won't be any pre processor.

Reply:
Ok, I got confused by the fact that the variable is named `apiPostMessageProcessorChain` in the `AsyncApiReactor`.
I suggest to align the names.